graphics/surface.cpp

```diff
@@ -903,27 +901,57 @@ void Surface::ditherFloyd(const byte *srcPalette, int srcPaletteCount, Surface *
 		{ nullptr, nullptr, 0 }
 	};
 
-	for (int y = 0; y < h; y++) {
-		const byte *src = &tmpSurf[y * w * 3];
-		byte *dst = (byte *)dstSurf->getBasePtr(0, y);
+	if (isRGB332) {
```

Comment:
Is the dstPaletteCount zero in this case? If so, palette lookup could be handled first and the format could be checked in the else section.

Reply:
I don't think it makes sense as RGB332 is a faster path than palette lookup and we never need to combine them

Reply:
Sorry, I was unclear. I wasn't suggesting combining, just flipping the if / else. Checking the format would not happen unless there was no palette, in which case it could even be an error() if the format doesn't match RGB332

Reply:
Done